### ISSUE TITLE
Check option to build 'kie-tools' repository code with java 17

### ISSUE THREAD

**Issue by jomarko** [apache/incubator-kie-issues#40]
With #6, we should be able to build codebase using java 17 (see https://www.gwtproject.org/release-notes.html?#Release_Notes_2_10_0). We should investigate this to see:
- if it is possible
- what are build performance consequences

**Comment by tiagobento** [comment]
@jomarko I don't think we can evaluate building the Editors alone. There are several places depending on Java on 'kie-tools', so when a version is upgraded for a package, it is upgraded for all packages. That has been how we're operating to maintain cosistency and avoid some parta of the monorepo to rot. Can you please update this issue to a broader scope? Thanks!

**Comment by jomarko** [comment]
@tiagobento sure, just tell me what the broader scope is, is it whole `kie-tools`, whole `kiegroup` organization, .... ?

**Comment by tiagobento** [comment]
The `kie-tools` repository. I'd say, dependency-wise, it's the smallest non-divisible entity :)

**Comment by jomarko** [comment]
Doing some tests as https://github.com/jomarko/kie-tools/pull/5

**Comment by yesamer** [comment]
I would reduce the scope of this ticket, assigning java 11 and still not 17 as the minimal version for build time.
Almost all java modules are already using java 11 at build time, the only 2 missing are the stunner forks. 
However, I think this is not a high-priority issue, and it should be done if the required effort is not high: i.e. if this change requires massive architectural changes we can skip it IMHO.

**Comment by yesamer** [comment]
@tiagobento @jomarko According to the assessment I did for https://github.com/kiegroup/kie-tools/pull/1456, It's unlikely we'll be able to build stunner-editor in JDK 9+, due to partial compatibility with Java 9+.
The main friction is the `super-source` feature ([here](https://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideModuleXml) ), which conflicts with Java 9 modules system.
If you try to build with JDK 11, you would see messages like:
`Compilation failure.../Generated.java:[41,1] package exists in another module: java.compiler`

I talked with D. about this, GWT team is aware of the issue, but they will address that in J2CL only, according to his knowledge.

Based on this, I think that building our GWT modules with Java 9+ JDK is not realistic, therefore I would close this ticket and continue using Java 8 until we support this.
